src/neighborsearch/kball_searcher.jl

```diff
@@ -0,0 +1,52 @@
+# ------------------------------------------------------------------
+# Licensed under the ISC License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+KBallSearcher(object, neighborhood)
+
+A method for searching neighbors in spatial `object` inside `neighborhood`.
+Used when doing estimation combining k-nearest and radius.
+It searches first for the k-nearest by using the internal kdtree and then
+limiting the results within the `BallNeighborhood` by calling `isneighbor`
+"""
+struct KBallSearcher{S<:AbstractNeighborSearcher} <: AbstractBoundedNeighborSearcher
+  searcher::S
+  nmax::Int
+end
+
+object(searcher::KBallSearcher) = object(searcher.searcher)
+
+maxneighbors(searcher::KBallSearcher) = searcher.nmax
+
+function search!(neighbors::AbstractVector{Int}, xₒ::AbstractVector,
+  searcher::KBallSearcher; mask=nothing)
+
+  knn_searcher = searcher.searcher #internar nearest searcher
+  k = searcher.nmax
+  neigh = knn_searcher.neigh       #the Ball Neighborhood
+  object = knn_searcher.object
+  radius = neigh.radius
+
+  inds, dists = knn(knn_searcher.kdtree, xₒ, k, true)
```

Comment:
I think what you really want to do here is leverage the inner searcher and call the `search!` method for it?

```julia
locs = search(xₒ, searcher.searcher, mask=mask)
```
And then post-process the locations with `isneighbor`?

Reply:
We could do that, but in that way we would need to recalculate distances (search function does not return distances that has already done by the inner kdtree). That is the reason for implementing this as it is.

Reply:
What we can do then is consider `KBallSearcher` its own implementation without relying on other existing searchers. In this case, it could have two fields `ball` and `k`. The constructor instantiates the kdtree similar to what is done in the NearestNeighborSearcher constructor, and then we implement the search explicitly without an inner searcher. What do you think?

Reply:
Much better I think. For example:
` Kriging(
:cu => (variogram=vmodel,searcher=KBallSearcher(k=25,radius=50.0))
)`

Reply:
I think the Kriging interface is another issue that we could consider later. Let's first work on this PR to get the KBallSearcher in place, then we proceed to the other PR to figure out an optimal set of parameters 👍